GitUI/CommandsDialogs/FormResolveConflicts.cs

```diff
@@ -1053,16 +1224,75 @@ private void OpenMergeTool()
                 try
                 {
                     var items = GetConflicts();
+                    _conflictItemsCount = items.Count;
 
-                    StartProgressBarWithMaxValue(items.Count);
+                    List<ConflictData> filesDeletedLocallyAndModifiedRemotely = new List<ConflictData>();
+                    List<ConflictData> filesModifiedLocallyAndDeletedRemotely = new List<ConflictData>();
+                    List<ConflictData> filesRemained = new List<ConflictData>();
                     foreach (var conflictData in items)
+                    {
+                        if (string.IsNullOrEmpty(conflictData.Local.Filename) && !string.IsNullOrEmpty(conflictData.Remote.Filename))
+                        {
+                            filesDeletedLocallyAndModifiedRemotely.Add(conflictData);
+                        }
+                        else if (!string.IsNullOrEmpty(conflictData.Local.Filename) && string.IsNullOrEmpty(conflictData.Remote.Filename))
+                        {
+                            filesModifiedLocallyAndDeletedRemotely.Add(conflictData);
+                        }
+                        else
+                        {
+                            filesRemained.Add(conflictData);
+                        }
+                    }
+
+                    // This is needed the task dialog shows the same order of files as selected in the grid
+                    filesDeletedLocallyAndModifiedRemotely.Reverse();
```

Comment:
This isn't perf optimal - we create a list, add items and the reverse. Why not just add items in the order we need?
I.e. 
```cs
List<ConflictData> filesDeletedLocallyAndModifiedRemotely = new List<ConflictData>();
foreach (var conflictData in items)
{
    filesDeletedLocallyAndModifiedRemotely.Insert(0, conflictData);
}
```